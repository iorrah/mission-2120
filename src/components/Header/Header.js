import React from "react";
import "./Header.scss";

const Header = function() {
  return <header className="header">...</header>;
};

export default Header;
