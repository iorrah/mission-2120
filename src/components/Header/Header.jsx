import React from "react";

const Header = function() {
  return <header className="header">...</header>;
};

export default Header;
