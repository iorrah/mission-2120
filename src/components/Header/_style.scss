.header {
  padding: 0.5rem 1.2rem 0.5rem 0.8rem;
  background-color: $white;
  display: flex;
  justify-content: space-between;
  height: 3rem;
  overflow: hidden;
  z-index: 1;
  position: relative;

  .header__logo-container {
    display: inline-block;
    width: 1.8rem;
    margin: 0 2rem 0 0;
  }

  .header__logo {
    display: block;
    width: 100%;
  }

  .header__search-container {
    display: inline-block;
    width: 30rem;
    position: relative;
    top: -0.55rem;
  }

  .header__search-icon {
    position: absolute;
    width: 1rem;
    top: 0.65rem;
  }

  .header__input {
    border: 0;
    display: block;
    width: 100%;
    font-size: 0.8rem;
    padding: 0.5rem 0.5rem 0.5rem 1.7rem;

    &:focus {
      outline: none;
    }
  }

  .header__bell-container {
    display: inline-block;
    width: 1.7rem;
    margin: 0 2rem 0 0;
  }

  .header__bell {
    display: block;
    width: 100%;
    position: relative;
    top: -0.1rem;
  }

  .header__avatar-container {
    display: inline-block;
    width: 2rem;
  }

  .header__avatar {
    display: block;
    width: 100%;
    border-radius: 100%;
  }
}
