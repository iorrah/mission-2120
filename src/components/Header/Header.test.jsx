import React from "react";
import { render } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import Header from "./Header";

describe("Header", () => {
  test("should shallow correctly", () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  });

  test("should mount correctly", () => {
    expect(mount(<Header />)).toMatchSnapshot();
  });

  test("should render correctly", () => {
    expect(render(<Header />)).toMatchSnapshot();
  });

  it("renders the `.Header` div", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(".header").exists()).toBeTruthy();
  });

  it("renders the `.header__logo`", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(".header__logo").exists()).toBeTruthy();
  });

  it("renders the search `.header__input`", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(".header__input").exists()).toBeTruthy();
  });

  it("renders the notification `.header__bell`", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(".header__bell").exists()).toBeTruthy();
  });

  it("renders the user `.header__avatar`", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(".header__avatar").exists()).toBeTruthy();
  });
});
