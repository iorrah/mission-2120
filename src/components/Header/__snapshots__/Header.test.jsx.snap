// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should mount correctly 1`] = `ReactWrapper {}`;

exports[`Header should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header__left"
        >
          <div
            class="header__logo-container"
          >
            <img
              alt="Logo"
              class="header__logo"
              src="img/icon-1.png"
            />
          </div>
          <div
            class="header__search-container"
          >
            <img
              alt="Magnifying glass"
              class="header__search-icon"
              src="img/icon-12.png"
            />
            <input
              class="header__input"
              placeholder="Search"
              type="text"
            />
          </div>
        </div>
        <div
          class="header__right"
        >
          <div
            class="header__bell-container"
          >
            <img
              alt="Notifications"
              class="header__bell"
              src="img/icon-11.png"
            />
          </div>
          <div
            class="header__avatar-container"
          >
            <img
              alt="Current user"
              class="header__avatar"
              src="img/avatar.png"
            />
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header__left"
      >
        <div
          class="header__logo-container"
        >
          <img
            alt="Logo"
            class="header__logo"
            src="img/icon-1.png"
          />
        </div>
        <div
          class="header__search-container"
        >
          <img
            alt="Magnifying glass"
            class="header__search-icon"
            src="img/icon-12.png"
          />
          <input
            class="header__input"
            placeholder="Search"
            type="text"
          />
        </div>
      </div>
      <div
        class="header__right"
      >
        <div
          class="header__bell-container"
        >
          <img
            alt="Notifications"
            class="header__bell"
            src="img/icon-11.png"
          />
        </div>
        <div
          class="header__avatar-container"
        >
          <img
            alt="Current user"
            class="header__avatar"
            src="img/avatar.png"
          />
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header should shallow correctly 1`] = `ShallowWrapper {}`;
